src/keyboardView.js

```diff
@@ -0,0 +1,293 @@
+        ["Ctrl", "Alt", " ", "alt", "En", "←", "↓", "→"],
+      ],
+      arrCode: [
+        ["Backquote", "Digit1", "Digit2", "Digit3", "Digit4", "Digit5", "Digit6", "Digit7", "Digit8", "Digit9", "Digit0", "Minus", "Equal", "Backspace"],
+        ["Tab", "KeyQ", "KeyW", "KeyE", "KeyR", "KeyT", "KeyY", "KeyU", "KeyI", "KeyO", "KeyP", "BracketLeft", "BracketRight", "Backslash"],
+        ["CapsLock", "KeyA", "KeyS", "KeyD", "KeyF", "KeyG", "KeyH", "KeyJ", "KeyK", "KeyL", "Semicolon", "Quote", "Enter"],
+        ["ShiftLeft", "KeyZ", "KeyX", "KeyC", "KeyV", "KeyB", "KeyN", "KeyM", "Comma", "Period", "Slash", "ArrowUp", "ShiftRight"],
+        ["ControlLeft", "AltLeft", "Space", "AltRight", "MetaRight", "ArrowLeft", "ArrowDown", "ArrowRight"],
+      ],
+    };
+  }
+
+  bindEvents() {
+    this.wrapper.addEventListener("mousedown", (e) => this.onClickHandle(e));
+    this.wrapper.addEventListener("mouseup", (e) => this.onMouseUp(e));
+    document.addEventListener("keydown", (e) => this.changeLanguageOnShortCut(e));
+    document.addEventListener("keydown", (e) => this.onKeyDown(e));
+    document.addEventListener("keyup", (e) => this.onKeyUp(e));
+  }
+
+  removeShiftOnClick(e) {
+    if ((e.target.id === "ShiftLeft" || e.target.id === "ShiftRight") && e.target.classList.contains("active")) {
+      this.isNotShift();
+    }
+  }
+
+  addShiftClick(e) {
+    if ((e.target.id === "ShiftLeft" || e.target.id === "ShiftRight") && !e.target.classList.contains("active")) {
+      this.isShift();
+      document.querySelector("#Shift").classList.add("active");
+    }
+  }
+
+  changeLang(language, keys) {
+    const keyboard = document.querySelector(".key-overlay");
+    this.language = language;
+    localStorage.removeItem("language");
+    localStorage.setItem("language", language);
+    keyboard.remove();
+    this.renderKeys(keys);
+  }
+
+  changeLanguageOnShortCut(e) {
+    e.preventDefault();
+    if (e.key === "Control" && e.altKey) {
+      if (this.language === "en" && !e.repeat) {
+        this.changeLang("ru", this.keys.keyArrRu);
+        document.querySelector("#AltLeft").classList.add("active");
+      } else if (this.language === "ru" && !e.repeat) {
+        this.changeLang("en", this.keys.keyArrEn);
+        document.querySelector("#AltLeft").classList.add("active");
+      }
+    }
+  }
+
+  removeUpperCase() {
+    this.key.forEach((el) => {
+      el.classList.remove("upper");
+    });
+  }
+
+  setCapsState(e) {
+    if (e.key === "CapsLock") {
+      this.setUpperKeys();
+      if (!document.querySelector("#CapsLock").classList.contains("active")) {
+        this.removeUpperCase();
+      }
+    }
+  }
+
+  setUpperKeys() {
+    this.key.forEach((el) => {
+      if (el.innerText.length === 1) {
+        el.classList.add("upper");
+      }
+    });
+  }
+
+  onShiftEnd(e) {
+    if (e.code === "ShiftLeft" || e.code === "ShiftRight") {
+      this.isNotShift();
+    }
+  }
+
+  isShift() {
```

Comment:
when you create a function with a name starting from 'is...', it is mean function return a boolean value

Reply:
Fixed